eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/ChannelRelayer.scala

```diff
@@ -90,35 +91,33 @@ object ChannelRelayer {
             replyTo ! Relayer.OutgoingChannels(selected.toSeq)
             Behaviors.same
 
-          case WrappedLocalChannelUpdate(LocalChannelUpdate(_, channelId, shortChannelId, remoteNodeId, _, channelUpdate, commitments)) =>
-            context.log.debug(s"updating local channel info for channelId=$channelId shortChannelId=$shortChannelId remoteNodeId=$remoteNodeId channelUpdate={} commitments={}", channelUpdate, commitments)
+          case WrappedLocalChannelUpdate(lcu@LocalChannelUpdate(_, channelId, shortIds, remoteNodeId, _, channelUpdate, commitments)) =>
+            context.log.debug(s"updating local channel info for channelId=$channelId realScid=${shortIds.real} localAlias=${shortIds.localAlias} remoteNodeId=$remoteNodeId channelUpdate={} commitments={}", channelUpdate, commitments)
             val prevChannelUpdate = channels.get(channelId).map(_.channelUpdate)
             val channel = Relayer.OutgoingChannel(remoteNodeId, channelUpdate, prevChannelUpdate, commitments)
             val channels1 = channels + (channelId -> channel)
-            val scid2channels1 = scid2channels + (channelUpdate.shortChannelId -> channelId)
+            val mappings = lcu.scidsForRouting.map(_ -> channelId).toMap
+            context.log.debug("adding mappings={} to channelId={}", mappings.keys.mkString(","), channelId)
+            val scid2channels1 = scid2channels ++ mappings
             val node2channels1 = node2channels.addOne(remoteNodeId, channelId)
             apply(nodeParams, register, channels1, scid2channels1, node2channels1)
 
-          case WrappedLocalChannelDown(LocalChannelDown(_, channelId, shortChannelId, remoteNodeId)) =>
-            context.log.debug(s"removed local channel info for channelId=$channelId shortChannelId=$shortChannelId")
+          case WrappedLocalChannelDown(LocalChannelDown(_, channelId, shortIds, remoteNodeId)) =>
+            context.log.debug(s"removed local channel info for channelId=$channelId localAlias=${shortIds.localAlias}")
             val channels1 = channels - channelId
-            val scid2Channels1 = scid2channels - shortChannelId
+            val scid2Channels1 = scid2channels - shortIds.localAlias
```

Comment:
Good catch. I've done a minimalist cleanup in a1ebd760e, but it's not possible to test as-is (because we do clean-up the `channels` map, which will cause relays to fail).

I started to add a `GetData` method to `ChannelRelayer`, that would return the whole state, but it's not trivial because one of the maps is mutable, making it unsafe to share with the outside. So I rolled back, LMKWYT.

Reply:
That's good enough :+1: